src/odemis/gui/cont/streams.py

```diff
@@ -153,6 +154,20 @@ def __init__(self, stream_bar, stream, tab_data_model, show_panel=True, view=Non
             if hasattr(self.stream, "selectionWidth"):
                 self._add_selwidth_ctrl()
                 
+        if hasattr(self.tab_data_model, "roa"):
```

Comment:
That's the right direction, but it's still not _enough_! Let's move entirely the handling of `.repetition` et al. here. In the same way that the blocks above have `if hasattr(stream, "spectrumBandwidth")` and other VAs, you can write `if hasattr(stream, "repetition"):`, and call a (new) method, which will add the repetition control, pixelSize control and connect the ROA.